lti_consumer/tests/unit/plugin/test_views.py

```diff
@@ -25,6 +25,30 @@
 from lti_consumer.lti_xblock import LtiConsumerXBlock
 from lti_consumer.lti_1p3.tests.utils import create_jwt
 from lti_consumer.tests.unit.test_utils import make_xblock
+from lti_consumer.plugin.views import parse_custom_parameters
+
+
+class TestParseCustomParameters(TestCase):
+    """
+    Test `parse_custom_parameters` function.
+    """
+
+    @patch('lti_consumer.plugin.views.resolve_custom_parameter_template')
+    def test_with_custom_parameters(self, mock_resolve_custom_parameter_template):
+        """
+        Check with custom parameters.
+        """
+        block = Mock()
+        block.custom_parameters = ['test=test']
+
+        self.assertEqual(parse_custom_parameters(block), {'test': 'test'})
+        mock_resolve_custom_parameter_template.assert_not_called()
+
+        mock_resolve_custom_parameter_template.return_value = ''
+        block.custom_parameters = ['test=${test}']
+
+        self.assertEqual(parse_custom_parameters(block), {'test': ''})
+        mock_resolve_custom_parameter_template.assert_called_once_with(block, '${test}')
```

Comment:
I think this is equivalent.  @kuipumu 
```suggestion
        block.custom_parameters = ['test=test', 'test2=${test}']
        mock_resolve_custom_parameter_template.return_value = ''
        
        self.assertDictEqual(parse_custom_parameters(block), {'test1': 'test', 'test2':''})
        mock_resolve_custom_parameter_template.assert_called_once_with(block, '${test}')
```

In case you would want to add more than 2 custom parameters, we should start using `mock_calls` to check what the `mock_resolve_custom_parameter_template` was called with, and to start using `side_effect` to return a different element of an iterable every time `mock_resolve_custom_parameter_template` is called.